org-brain.el

```diff
@@ -1352,6 +1387,104 @@ After refiling, all headlines will be given an id."
+        (setq org-brain-pins (mapcar #'replace-entry org-brain-pins))
+        (setq org-brain--vis-history (mapcar #'replace-entry org-brain--vis-history))
+        (setq org-brain-selected (mapcar #'replace-entry org-brain-selected)))
+      (when (member parent
+                    (org-brain--linked-property-entries new-entry "BRAIN_PARENTS"))
+          (org-brain-remove-relationship parent new-entry))
+      (org-save-all-org-buffers)
+      (when (eq major-mode 'org-brain-visualize-mode)
+        (org-brain-visualize new-entry))
+      new-entry)))
+
+;;;###autoload
+(defun org-brain-change-local-parent (entry parent)
+  "Refile ENTRY to be a local child of PARENT.
+Entries are relinked so existing parent-child relationships are unaffected.
+
+If called interactively, ENTRY is the current entry
+and PARENT is prompted for among the list of ENTRY's linked parents.
+Returns the new refiled entry."
+  (interactive
+   (let* ((this-entry (org-brain-entry-at-pt))
+          (linked-parents (org-brain--linked-property-entries this-entry "BRAIN_PARENTS"))
+          (chosen-parent (org-brain-choose-entry "Refile to parent: " linked-parents)))
+     (list this-entry chosen-parent)))
+  (let ((old-parent (car (org-brain--local-parent entry)))
+        (new-entry (org-brain-refile-to entry parent)))
+    (org-brain-add-relationship old-parent new-entry)
+    (org-brain--revert-if-visualizing)
+    new-entry))
+
+;;;###autoload
+(defun org-brain-unlink-child (entry child)
+  "Remove CHILD as a child of ENTRY.
+If CHILD is a local child of ENTRY, CHILD is refiled to another parent."
+  (interactive (let ((e (org-brain-entry-at-pt)))
+                 (list e (org-brain-choose-entry "Unlink child: "
+                                                 (org-brain-children e)
+                                                 nil t))))
+  (org-brain-unlink-parent child entry))
+
+;;;###autoload
+(defun org-brain-unlink-parent (entry parent)
+  "Remove PARENT as a parent of ENTRY.
+If ENTRY is a local child of PARENT,
+ENTRY is refiled to another parent, chosen by the function."
+  (interactive (let ((e (org-brain-entry-at-pt)))
+                 (list e (org-brain-choose-entry "Unlink parent: "
+                                                 (org-brain-parents e)
+                                                 nil t))))
+  (when (member parent (org-brain--local-parent entry))
+    (let* ((linked-parents (org-brain--linked-property-entries entry "BRAIN_PARENTS"))
+           (headline-parents (seq-filter (lambda (e) (not (org-brain-filep e))) linked-parents))
+           (file-parents (seq-filter 'org-brain-filep linked-parents))
+           (new-parent (car (append headline-parents file-parents))))
+      (when (null new-parent)
+        (error "Cannot unlink a headline entry's last parent"))
+      (setq entry (org-brain-change-local-parent entry new-parent))))
+  (org-brain-remove-relationship parent entry)
+  (org-brain--revert-if-visualizing))
+
```

Comment:
I tried to modify `org-brain-remove-child/parent` as discussed, and the tests I've made seems to be working. This is the suggested code:

```
(defun org-brain-remove-child (entry child)
  "Remove CHILD from ENTRY.
If called interactively use `org-brain-entry-at-point' and prompt for CHILD."
  (interactive (let ((e (org-brain-entry-at-pt)))
                 (list e (org-brain-choose-entry "Remove child: "
                                                 (org-brain-children e)
                                                 nil t))))
  (if (member child (org-brain--local-children entry))
      (if (and (> (length (org-brain-parents child)) 1)
               (y-or-n-p
                (format "%s is %s's local parent. Would you like to change the local parent of %s? "
                        (org-brain-title entry) (org-brain-title child) (org-brain-title child))))
          (let* ((linked-parents (org-brain--linked-property-entries child "BRAIN_PARENTS"))
                 (new-parent (if (equal 1 (length linked-parents))
                                 (car-safe linked-parents)
                               (org-brain-choose-entry "Refile to parent: " linked-parents))))
            (org-brain-remove-relationship entry (org-brain-change-local-parent child new-parent)))
        (org-brain-delete-entry child))
    (org-brain-remove-relationship entry child))
  (org-brain--revert-if-visualizing))

(defun org-brain-remove-parent (entry parent)
  "Remove PARENT from ENTRY.
If called interactively use `org-brain-entry-at-pt' and prompt for PARENT."
  (interactive (let ((e (org-brain-entry-at-pt)))
                 (list e (org-brain-choose-entry "Remove parent: "
        	                                 (org-brain-parents e)
	                                         nil t))))
  (if (member entry (org-brain--local-children parent))
      (if-let* ((linked-parents (org-brain--linked-property-entries entry "BRAIN_PARENTS"))
                (new-parent (if (equal 1 (length linked-parents))
                                (car-safe linked-parents)
                              (org-brain-choose-entry (format "Removing %s's local parent. Refile to: "
                                                              (org-brain-title entry))
                                                      linked-parents))))
          (org-brain-remove-relationship parent (org-brain-change-local-parent entry new-parent))
        (error "%s is %s's only parent, it can't be removed"
               (org-brain-title parent) (org-brain-title entry)))
    (org-brain-remove-relationship parent entry))
  (org-brain--revert-if-visualizing))
```

If making these changes, I don't think we need `org-brain-unlink-child/parent`?

Reply:
Can we just define `org-brain-remove-parent` in terms of `org-brain-remove-child`? I.e.,

```
(defun org-brain-remove-parent (entry parent)
  "Remove PARENT from ENTRY.
If called interactively use `org-brain-entry-at-pt' and prompt for PARENT."
  (interactive (let ((e (org-brain-entry-at-pt)))
                 (list e (org-brain-choose-entry "Remove parent: "
        	                                 (org-brain-parents e)
	                                         nil t))))
  (org-brain-remove-child parent entry))
 ```

Reply:
Ah I see — we want `remove-parent` to error out rather than deleting the current entry. Makes sense.